WordPress/openverse
# Update Playwright to use a custom test fixture with global mocks

Creates a `test` fixture which overrides the base Playwright test fixture. This allows for easily setting global route mocks and other settings. Most crucially it removes the thousands of logs cluttering the output about the failed proxying of the analytics server.

This PR also makes our provider mocks global and adds a global mock for the analytics event endpoint, to prevent that from outputting hundreds of proxy-related error logs during the run of the test suite.

`frontend/test/playwright/utils/test-fixture.ts` is the main thing to review here. The rest is just implementations of the new fixture (switching the import from playwright to our custom test fixture).

~This will be cleaned up and undrafted once #2194 is merged.~

## Testing instructions

- Observe tests are still working and logs are no longer cluttered with analytics event warnings
- Ensure provider APIs are still mocked correctly in all relevant tests
- Try to `import { test } from "@playwright/test"` and you should be able to observe a lint error.
